### PR TITLE
add SPDX tags and github CI actions

### PR DESCRIPTION
**Pull request by lsf37**
Link-check is for now excluded, because that should be done on the generated site.

### CODE REVIEW

**Comment by lsf37** [comment]
@kent-mcleod @xurtis what do you think of the way [here](https://github.com/seL4/docs/pull/67/files#diff-44e1904127d1523ee8db9b56a236ee600d3900bcec03d5b93e16e8daced74707) to add SPDX license info to markdown files that have meta-info at the top?

It basically creates meta info fields for license and copyright. We don't really need them for any processing, but jekyll doesn't like comments at the very top of the file and it avoids an awkward in-between copyright header.

**Comment by xurtis** [comment]
That strategy seems fine for jekyll content files. I guess all we really require is that the license appears in the file at all, doesn't need to be a comment. Having it in metadata might be a plus, if anything.

**Comment by lsf37** [comment]
> That strategy seems fine for jekyll content files. I guess all we really require is that the license appears in the file at all, doesn't need to be a comment. Having it in metadata might be a plus, if anything.

Cool, that mirrors my thinking. @kent-mcleod you happy as well?

**Comment by kent-mcleod** [comment]
Works for me.

**Comment by lsf37** [comment]
Cool, will start adding the rest!

File: Hardware/CEI_TK1_SOM/L4TCan/tegra124-tk1-som-gpio-default.dtsi [hunk 1]
```diff
@@ -1,79 +1,81 @@
...
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+/* SPDX-License-Identifier: GPL-2.0-or-later */
```
Comment (lsf37) [inline Hardware/CEI_TK1_SOM/L4TCan/tegra124-tk1-som-gpio-default.dtsi:19]: This is the only real change in this file -- the rest is line-ending conversion from CRLF to LF to calm down the trailing whitespace check.

**Comment by lsf37** [comment]
This is now ready for review. Might be easier to do by commit (there is one very big SPDX commit, and a few smaller ones for python style and github actions).

I'm planning to ignore the style test failure for the C file, because it's a 3rd-party file.

**Comment by xurtis** [comment]
> I'm planning to ignore the style test failure for the C file, because it's a 3rd-party file.

I think there's a [`.licenseignore`](https://github.com/seL4/util_libs/blob/master/.licenseignore) mechanism in the license checker that might be best to use for that.

**Comment by lsf37** [comment]
> I think there's a [`.licenseignore`](https://github.com/seL4/util_libs/blob/master/.licenseignore) mechanism in the license checker that might be best to use for that.

The failure is in the style checker (not license), but that one also has an ignore mechanism, which I should indeed check out.

**Review by xurtis — APPROVED** [review]
LGTM

**Comment by lsf37** [comment]
I should probably add here for posterity that I marked most things as "copyright seL4 foundation" where it implicitly was NICTA/D61 before, since this repo has been officially handed over. The foundation usually doesn't insist on owning the copyright for contributions, this was mostly just to keep things consistent and simple.

**Pull request merged**